packages/rancher-monitoring/rancher-monitoring.patch

```diff
@@ -1544,7 +1544,100 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/val
+   ##
++  ## Example Slack Config
++  ## config:
++  ##   route:
++  ##     group_by: ['job']
++  ##     group_wait: 30s
++  ##     group_interval: 5m
++  ##     repeat_interval: 3h
++  ##     receiver: 'slack-notifications'
++  ##   receivers:
++  ##   - name: 'slack-notifications'
++  ##     slack_configs:
++  ##     - send_resolved: true
++  ##       text: '{{ template "slack.rancher.text" . }}'
++  ##       api_url: <slack-webhook-url-here>
++  ##   templates:
++  ##   - /etc/alertmanager/config/*.tmpl
+   config:
+     global:
+       resolve_timeout: 5m
+@@ -145,6 +364,8 @@
+         receiver: 'null'
+     receivers:
+     - name: 'null'
++    templates:
++    - /etc/alertmanager/config/*.tmpl
+ 
+   ## Pass the Alertmanager configuration directives through Helm's templating
+   ## engine. If the Alertmanager configuration contains Alertmanager templates,
+@@ -160,25 +381,41 @@
+   ## ref: https://prometheus.io/docs/alerting/notifications/
+   ##      https://prometheus.io/docs/alerting/notification_examples/
+   ##
+-  templateFiles: {}
+-  #
+-  ## An example template:
+-  #   template_1.tmpl: |-
+-  #       {{ define "cluster" }}{{ .ExternalURL | reReplaceAll ".*alertmanager\\.(.*)" "$1" }}{{ end }}
+-  #
+-  #       {{ define "slack.myorg.text" }}
+-  #       {{- $root := . -}}
+-  #       {{ range .Alerts }}
+-  #         *Alert:* {{ .Annotations.summary }} - `{{ .Labels.severity }}`
+-  #         *Cluster:*  {{ template "cluster" $root }}
+-  #         *Description:* {{ .Annotations.description }}
+-  #         *Graph:* <{{ .GeneratorURL }}|:chart_with_upwards_trend:>
+-  #         *Runbook:* <{{ .Annotations.runbook }}|:spiral_note_pad:>
+-  #         *Details:*
+-  #           {{ range .Labels.SortedPairs }} • *{{ .Name }}:* `{{ .Value }}`
+-  #           {{ end }}
+-  #       {{ end }}
+-  #       {{ end }}
++  templateFiles:
++    rancher_defaults.tmpl: |-
++        {{- define "slack.rancher.text" -}}
++        {{ template "rancher.text_multiple" . }}
++        {{- end -}}
++
++        {{ define "cluster" }}{{ .ExternalURL | reReplaceAll ".*alertmanager\\.(.*)" "$1" }}{{ end }}
++
++        {{- define "rancher.text_multiple" }}
++        *Cluster:*  {{ template "cluster" . }}
++        {{- range .Alerts }}
++        {{ template "rancher.text_single" . }}
++        {{- end }}
++        {{- end }}
++
++        {{- define "rancher.text_single" -}}
++        {{- if .Labels.alertname }}
++        *Alert*: {{ .Labels.alertname }}
++        {{- end }}
++        {{- if .Annotations.summary }}
++        *Summary:* {{ .Annotations.summary }}
++        {{- end }}
++        {{- if .Labels.severity }}
++        *Severity:* `{{ .Labels.severity }}`
++        {{- end }}
++        {{- if .Annotations.description }}
++        *Description:* {{ .Annotations.description }}
++        {{- end }}
++        {{- if .Annotations.runbook }}
++        *Runbook:* <{{ .Annotations.runbook }}|:spiral_note_pad:>
++        {{- end }}
++        *Details:*
```

Comment:
No if check here ? To see if Labels.sortedPairs is empty.

Reply:
Good catch! Updated in latest commit